### PR TITLE
Remove build qualifier from server's Version 

### PR DESCRIPTION
**Pull request by alpar-t**
Build on #35155 ( and contains commits from )

- Rename the constant with an automated IDE refactoring - this generates a high number of changes and changed files. 
- Pass only the numeric part of the version in places like plugin properties so the server doesn't have to deal with qualifier and snapshot. 
- adds a new substitution to doc tests as the docs now need to know about both.
- adds a new `String` version field to `Build` to hold the complete version string passed on from the build.
- replace `Version.getDisplayVersion`  with `Build.CURRENT.getVersion()` introduced above
- add a new `build_version` parameter to `version`. Was not sure this is the way to go, feedback is welcome. The alternative is to have `version.number` show the build qualifier as well without adding a new `build_version`. It did use to have `-SNAPSHOT` in there, but it's not technically a "number". Not that the content of `version.number` will change in regardless. 

With this change elasticsearch greets as: 
```
{
  "name" : "node-0",
  "cluster_name" : "distribution_run",
  "cluster_uuid" : "qvtY3T1bQi-vR4zO4lrpyQ",
  "version" : {
    "number" : "7.0.0",
    "build_flavor" : "default",
    "build_type" : "zip",
    "build_hash" : "26be21b",
    "build_date" : "2018-11-05T13:40:03.172633Z",
    "build_snapshot" : true,
    "build_version" : "7.0.0-alpha1-SNAPSHOT",
    "lucene_version" : "8.0.0",
    "minimum_wire_compatibility_version" : "6.6.0",
    "minimum_index_compatibility_version" : "6.0.0-beta1"
  },
  "tagline" : "You Know, for Search"
}
```

And in the logs we have: 
```
 version [7.0.0-alpha1-SNAPSHOT] is a pre-release version of Elasticsearch and is not suitable for production
```

### CODE REVIEW

**Review by nik9000 — CHANGES_REQUESTED** [review]
OK! I left a few small things. I like `Build.getVersion()` - though maybe not the name? Maybe `Build.getQualifiedVersion` or something. And it should have javadoc. Because we already have `Version` and it isn't the same thing.

**Comment by atorok** [comment]
Thanks for your review @nik9000 . I made changes as suggested.

File: qa/wildfly/src/test/java/org/elasticsearch/wildfly/WildflyIT.java [hunk 1]
```diff
@@ -54,14 +53,16 @@
...
         try (CloseableHttpClient client = HttpClientBuilder.create().build()) {
             final String str = String.format(
-                    Locale.ROOT,
-                    "http://localhost:%d/wildfly-%s%s/transport/employees/1",
-                    Integer.parseInt(System.getProperty("tests.jboss.http.port")),
-                    Version.CURRENT,
-                    Build.CURRENT.isSnapshot() ? "-SNAPSHOT" : "");
+                Locale.ROOT,
+                "%s/employees/1",
+                System.getProperty("tests.jboss.home")
```
Comment (nik9000) [inline qa/wildfly/src/test/java/org/elasticsearch/wildfly/WildflyIT.java]: I think "home" means something else in this context. Maybe `test.wildfly.root`?

File: server/src/main/java/org/elasticsearch/Build.java [hunk 2]
```diff
@@ -221,6 +243,13 @@ public static void writeBuild(Build build, StreamOutput out) throws IOException
         out.writeString(build.shortHash());
         out.writeString(build.date());
         out.writeBoolean(build.isSnapshot());
+        if (out.getVersion().onOrAfter(Version.V_7_0_0)) {
+            out.writeString(build.getQualifiedVersion());
+        }
+    }
+
+    public String getQualifiedVersion() {
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/Build.java]: Could you add javadoc to this?

File: server/src/main/java/org/elasticsearch/Build.java [hunk 3]
```diff
@@ -235,9 +264,13 @@ public boolean isSnapshot() {
         return isSnapshot;
     }
 
+    public boolean isProductionRelease() {
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/Build.java]: This too?

File: server/src/main/java/org/elasticsearch/bootstrap/Elasticsearch.java [hunk 4]
```diff
@@ -102,7 +101,7 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
             final String versionOutput = String.format(
                     Locale.ROOT,
                     "Version: %s, Build: %s/%s/%s/%s, JVM: %s",
-                    Version.displayVersion(Version.CURRENT, Build.CURRENT.isSnapshot()),
+                Build.CURRENT.getQualifiedVersion(),
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/bootstrap/Elasticsearch.java]: I think the indentation on this one is off.

File: server/src/main/java/org/elasticsearch/node/Node.java [hunk 5]
```diff
@@ -577,11 +576,11 @@ protected Node(
         }
     }
 
-    static void warnIfPreRelease(final Version version, final boolean isSnapshot, final Logger logger) {
-        if (!version.isRelease() || isSnapshot) {
+    static void warnIfPreRelease(final Build build,final Logger logger) {
+        if (build.isProductionRelease() == false) {
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/node/Node.java]: I think maybe this isn't worth being its own method any more. Now that we're testing `isProductionRelease` instead.

**Review by nik9000 — APPROVED** [review]
I left a few super minor things but this LGTM. Good luck on the backport!

**Comment by atorok** [comment]
@nik9000 this is not planned to be back-ported. I have done so with the work leading up here, but don't plan to do it with this one. Do you think we should ?

**Comment by atorok** [comment]
After looking more into it, it seems that it was a deliberate decision not to have `-SNAPSHOT` in `version.number` so we'll probably want to stick to that.

**Comment by atorok** [comment]
@javanna that was done on purpose. Qualified builds will be tagged as such as needed.

**Pull request merged**